File: apps/Applications/ios/Local For Vocal/Local For Vocal/Sources/Views/Components/BudgetBuysView.swift
import SwiftUI

struct BudgetBuysView: View {
    // Props
    var title: String = "Budget Buys"
    var headerActionUrl: String?
    var items: [BudgetItem] = []

    struct BudgetItem: Identifiable, Decodable {
        let id: String
        let image: String
        let price: String
        let actionUrl: String?

        var safeId: String { id }
    }

    let columns = [
        GridItem(.flexible(), spacing: 16),
        GridItem(.flexible(), spacing: 16),
    ]

    var body: some View {
        VStack(alignment: .leading, spacing: 12) {
            // Header
            Button(action: {
                if let action = headerActionUrl {
                    AppLogger.debug("Navigate to: \(action)")
                }
            }) {
                HStack(spacing: 4) {
                    Text(title)
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(Color(hex: "#000000"))
                    Text("›")
                        .font(.system(size: 18, weight: .bold))  // Match title weight
                        .foregroundColor(Color(hex: "#000000"))
                }
            }
            .buttonStyle(PlainButtonStyle())
            .padding(.horizontal, 16)

            // Grid
            LazyVGrid(columns: columns, spacing: 16) {
                ForEach(items) { item in
                    BudgetBuysCard(item: item)
                }
            }
            .padding(.horizontal, 16)
        }
        .padding(.vertical, 16)
        .background(Color.white)
    }
}

struct BudgetBuysCard: View {
    let item: BudgetBuysView.BudgetItem

    var body: some View {
        Button(action: {
            if let action = item.actionUrl {
                AppLogger.debug("Navigate to: \(action)")
            }
        }) {
            ZStack {
                // Background Image
                if let url = URL(string: item.image) {
                    AsyncImage(url: url) { image in
                        image
                            .resizable()
                            .aspectRatio(contentMode: .fill)
                    } placeholder: {
                        Color(hex: "#F0F0F0")
                    }
                } else {
                    Color(hex: "#F0F0F0")
                }

                // Overlay
                VStack(spacing: -4) {
                    Text("UNDER")
                        .font(.custom("Didot", size: 18))  // Native uses Didot on iOS
                        .fontWeight(.medium)
                        .tracking(2)  // letterSpacing
                        .foregroundColor(.white)

                    Text("₹\(item.price)")
                        .font(.custom("Didot", size: 42))  // Native uses Didot
                        .fontWeight(.bold)
                        .foregroundColor(.white)
                }
            }
            .aspectRatio(1, contentMode: .fit)
            .background(Color(hex: "#F0F0F0"))
            .cornerRadius(12)
            .clipped()
        }
    }
}

File: apps/Applications/android/app/src/main/java/com/localforvocalstartup/app/ui/components/BudgetBuysView.kt
package com.localforvocalstartup.app.ui.components

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import coil.compose.AsyncImage

data class BudgetItem(
    val id: String,
    val image: String,
    val price: String,
    val actionUrl: String?
)

@Composable
fun BudgetBuysView(
    title: String = "Budget Buys",
    headerActionUrl: String? = null,
    items: List<BudgetItem> = emptyList(),
    onNavigate: (String) -> Unit = {}
) {
    Column(
        modifier = Modifier
            .fillMaxWidth()
            .background(Color.White)
            .padding(vertical = 16.dp),
        verticalArrangement = Arrangement.spacedBy(12.dp)
    ) {
        // Header
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .clickable { headerActionUrl?.let { onNavigate(it) } }
                .padding(horizontal = 16.dp),
            verticalAlignment = Alignment.CenterVertically,
            horizontalArrangement = Arrangement.spacedBy(4.dp)
        ) {
            Text(
                text = title,
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = Color.Black
            )
            Text(
                text = "›",
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = Color.Black
            )
        }

        // Custom 2-column Grid
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .padding(horizontal = 16.dp),
            verticalArrangement = Arrangement.spacedBy(16.dp)
        ) {
            val chunkedItems = items.chunked(2)
            for (rowItems in chunkedItems) {
                Row(
                    modifier = Modifier.fillMaxWidth(),
                    horizontalArrangement = Arrangement.spacedBy(16.dp)
                ) {
                    for (item in rowItems) {
                        BudgetBuysCard(
                            item = item,
                            modifier = Modifier.weight(1f),
                            onClick = { item.actionUrl?.let { onNavigate(it) } }
                        )
                    }
                    if (rowItems.size == 1) {
                        Spacer(modifier = Modifier.weight(1f))
                    }
                }
            }
        }
    }
}

@Composable
private fun BudgetBuysCard(
    item: BudgetItem,
    modifier: Modifier = Modifier,
    onClick: () -> Unit
) {
    Box(
        modifier = modifier
            .aspectRatio(1f)
            .clip(RoundedCornerShape(12.dp))
            .background(Color(0xFFF0F0F0))
            .clickable { onClick() },
        contentAlignment = Alignment.Center
    ) {
        AsyncImage(
            model = item.image,
            contentDescription = null,
            contentScale = ContentScale.Crop,
            modifier = Modifier.fillMaxSize()
        )

        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            verticalArrangement = Arrangement.spacedBy((-4).dp)
        ) {
            Text(
                text = "UNDER",
                fontFamily = FontFamily.Serif,
                fontSize = 18.sp,
                fontWeight = FontWeight.Medium,
                letterSpacing = 2.sp,
                color = Color.White
            )
            Text(
                text = "₹${item.price}",
                fontFamily = FontFamily.Serif,
                fontSize = 42.sp,
                fontWeight = FontWeight.Bold,
                color = Color.White
            )
        }
    }
}
